controllers/datadogagent/override/global.go

```diff
@@ -6,16 +6,212 @@
+	// Tags contains a list of tags to attach to every metric, event and service check collected.
+	if config.Tags != nil {
+		tags, _ := json.Marshal(config.Tags)
+		manager.EnvVar().AddEnvVar(&corev1.EnvVar{
+			Name:  apicommon.DDTags,
+			Value: string(tags),
+		})
+	}
+
+	// Provide a mapping of Kubernetes Labels to Datadog Tags.
+	if config.PodLabelsAsTags != nil {
+		podLabelsAsTags, _ := json.Marshal(config.PodLabelsAsTags)
+		manager.EnvVar().AddEnvVar(&corev1.EnvVar{
+			Name:  apicommon.DDPodLabelsAsTags,
+			Value: string(podLabelsAsTags),
+		})
+	}
+
+	// Provide a mapping of Kubernetes Annotations to Datadog Tags.
+	if config.PodAnnotationsAsTags != nil {
+		podAnnotationsAsTags, _ := json.Marshal(config.PodAnnotationsAsTags)
+		manager.EnvVar().AddEnvVar(&corev1.EnvVar{
+			Name:  apicommon.DDPodAnnotationsAsTags,
+			Value: string(podAnnotationsAsTags),
+		})
+	}
+
+	// NetworkPolicy contains the network configuration.
+	if config.NetworkPolicy != nil {
+		if apiutils.BoolValue(config.NetworkPolicy.Create) {
+			switch config.NetworkPolicy.Flavor {
+			case v2alpha1.NetworkPolicyFlavorKubernetes:
+				// node agent
+				_ = resourcesManager.NetworkPolicyManager().BuildAgentKubernetesNetworkPolicy(dda)
+				// dca
+				_ = resourcesManager.NetworkPolicyManager().BuildDCAKubernetesNetworkPolicy(dda)
+				// ccr
+				if apiutils.BoolValue(dda.Spec.Features.ClusterChecks.UseClusterChecksRunners) {
+					_ = resourcesManager.NetworkPolicyManager().BuildCCRKubernetesNetworkPolicy(dda)
+				}
+			case v2alpha1.NetworkPolicyFlavorCilium:
+				// TODO
+				// node agent
+				// dca
+				// ccr
+			}
+		}
+	}
 
-	// set image registry
+	// LocalService contains configuration to customize the internal traffic policy service.
+	gitVersion := resourcesManager.Store().GetVersionInfo()
+	// Service Internal Traffic Policy exists in Kube 1.21 but it is enabled by default since 1.22
+	if utils.IsAboveMinVersion(gitVersion, "1.21-0") {
+		if config.LocalService != nil {
+			if apiutils.BoolValue(config.LocalService.ForceEnableLocalService) || utils.IsAboveMinVersion(gitVersion, "1.22-0") {
+				if config.LocalService.NameOverride != nil {
+					_ = resourcesManager.ServiceManager().BuildAgentLocalService(dda, *config.LocalService.NameOverride)
+				} else {
+					_ = resourcesManager.ServiceManager().BuildAgentLocalService(dda, "")
+				}
+			}
+		}
+	}
+
+	// Kubelet contains the kubelet configuration parameters.
+	if config.Kubelet != nil {
+		var kubeletHostValueFrom *corev1.EnvVarSource
+		if config.Kubelet.Host != nil {
+			kubeletHostValueFrom = config.Kubelet.Host
+		} else {
+			kubeletHostValueFrom = &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{
+					FieldPath: apicommon.FieldPathStatusHostIP,
+				},
+			}
+		}
+		manager.EnvVar().AddEnvVar(&corev1.EnvVar{
+			Name:      apicommon.DDKubeletHost,
+			ValueFrom: kubeletHostValueFrom,
+		})
+
+		// tlsverify defaults to true in agent code
+		if !*config.Kubelet.TLSVerify {
```

Comment:
maybe we should explicitly set this to default True in the defaulting code, and then do
```suggestion
		if !apiutils.BoolValue(config.Kubelet.TLSVerify) {
```